### PR TITLE
New function alias: env()

### PR DESCRIPTION
**Pull request by kykyi**
In relation to https://github.com/casey/just/issues/1278, this PR adds a new `env` function which acts as an alias to both `env_var` and `env_var_or_default`:
```just
env_var('x') => env('x')
env_var_or_default('x', 42) => env('x', 42)
```

To make such a function signature possible, a new `UnaryOpt` enum variant was added which accounts for the majority of the touched files!

### CODE REVIEW

**Comment by casey** [comment]
Thanks for the PR! It looks like the `env` function will silently ignore excess arguments, which isn't great. How about something like:

```
  UnaryOpt(fn(&FunctionContext, &str, Option<&str>) -> Result<String, String>),
```

Which accepts only one or two arguments?

**Comment by kykyi** [comment]
Good call @casey! I have updated to introduce UnaryOpt 👌

File: src/parser.rs [hunk 1]
```diff
@@ -2334,6 +2334,34 @@ mod tests {
     },
   }
 
+  error! {
```
Comment (kykyi) [inline src/parser.rs:2336]: These tests break the `function_argument_<variant>` convention by adding a too-many and a too-few test, unsure if this is what you want or should just have the single test.

File: src/thunk.rs [hunk 2]
```diff
@@ -60,6 +67,18 @@ impl<'src> Thunk<'src> {
           arg: Box::new(arguments.pop().unwrap()),
           name,
         }),
+        (Function::UnaryOpt(function), 1..=2) => {
+          let a = Box::new(arguments.remove(0));
+          let opt_b = match arguments.pop() {
+            Some(value) => Box::new(Some(value)),
+            None => Box::new(None),
+          };
```
Comment (kykyi) [inline src/thunk.rs]: `remove(0)` to grab from the front of the list, then a `pop` on the last element, happy to hear a prettier way 😅

**Comment by casey** [comment]
Nice, merged!

**Pull request merged**